src/vs/workbench/contrib/positronPlots/browser/positronPlotsService.ts

```diff
@@ -745,17 +784,26 @@ export class PositronPlotsService extends Disposable implements IPositronPlotsSe
 			this._onDidSelectPlot.fire(plotClient.id);
 		}
 
+		// Dispose the plot client when this service is disposed (we own this
+		// object)
+		const disp = this._register(plotClient);
+
 		// Remove the plot from our list when it is closed
-		this._register(plotClient.onDidClose(() => {
-			const index = this._plots.indexOf(plotClient);
-			if (index >= 0) {
-				this._plots.splice(index, 1);
+		plotClient.register({
```

Comment:
I noticed some resource management issue here, we were never cleaning up event listeners. I've added a `register()` method on plot client instances so that we can tie these resources to the plot lifecycle.

This `register()` method is used in a more straightforward way below. Here is a bit more involved, we create a disposable object for our resource cleanup. I think that's a bit simpler than using an `onDidClose()` event whose event listener itself needs to be disposed of along with the client (and not the service as is currently the case).